packages/web-frontend/src/historyNavigation/historyMiddleware.js

```diff
@@ -12,92 +12,25 @@
 import {
   CLEAR_MEASURE,
   CLOSE_ENLARGED_DIALOG,
-  DIALOG_PAGE_ONE_TIME_LOGIN,
   GO_HOME,
-  openEnlargedDialog,
-  openUserPage,
   OPEN_ENLARGED_DIALOG,
-  setMeasure,
-  setOrgUnit,
-  setOverlayComponent,
-  setPasswordResetToken,
-  setVerifyEmailToken,
   SET_DASHBOARD_GROUP,
   SET_MEASURE,
   SET_ORG_UNIT,
   SET_PROJECT,
-  updateCurrentMeasureConfigOnceHierarchyLoads,
   updateHistoryLocation,
   UPDATE_MEASURE_CONFIG,
+  LOCATION_CHANGE,
 } from '../actions';
-import { DEFAULT_PROJECT_CODE } from '../defaults';
-import { setProject } from '../projects/actions';
 import { selectCurrentPeriodGranularity, selectMeasureBarItemById } from '../selectors';
-import { PASSWORD_RESET_PREFIX, URL_COMPONENTS, VERIFY_EMAIL_PREFIX } from './constants';
+import { URL_COMPONENTS } from './constants';
 import {
   addPopStateListener,
   attemptPushHistory,
   clearLocation,
-  getInitialLocation,
   setLocationComponent,
 } from './historyNavigation';
-import { PROJECTS_WITH_LANDING_PAGES, PROJECT_LANDING } from '../containers/OverlayDiv/constants';
-import { convertDateRangeToUrlPeriodString, decodeLocation } from './utils';
-
-export const reactToInitialState = store => {
```

Comment:
Move `reactToInitialState` and `reactToLocationChange` to sagas.js because they involve complex interaction with the store. This also allows us to re-use this behaviour to react to a location change in other flows of the app (eg. after logging in)

Reply:
Good call!